scripts/pre_commit_linter.py

```diff
@@ -630,1300 +876,1143 @@ def _lint_py_files(config_pylint, config_pycodestyle, files_to_lint, result):
+                    '%s   Import order checks failed' % _MESSAGE_TYPE_FAILED)
+                print summary_message
+                summary_messages.append(summary_message)
+            else:
+                summary_message = (
+                    '%s   Import order checks passed' % _MESSAGE_TYPE_SUCCESS)
+                print summary_message
+                summary_messages.append(summary_message)
 
-def _check_comments(all_files):
-    """This function ensures that comments follow correct style."""
-    print 'Starting comment checks'
-    print '----------------------------------------'
-    summary_messages = []
-    files_to_check = [
-        filename for filename in all_files if not
-        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
-        and filename.endswith('.py')]
-    message = 'There should be a period at the end of the comment.'
-    failed = False
-    space_regex = re.compile(r'^#[^\s].*$')
-    capital_regex = re.compile('^# [a-z][A-Za-z]* .*$')
-    with _redirect_stdout(_TARGET_STDOUT):
-        for filename in files_to_check:
-            file_content = FileCache.readlines(filename)
-            file_length = len(file_content)
-            for line_num in range(file_length):
-                line = file_content[line_num].strip()
-                next_line = ''
-                previous_line = ''
-                if line_num + 1 < file_length:
-                    next_line = file_content[line_num + 1].strip()
-                if line_num > 0:
-                    previous_line = file_content[line_num - 1].strip()
-
-                if line.startswith('#') and not next_line.startswith('#'):
-                    # Check that the comment ends with the proper punctuation.
-                    last_char_is_invalid = line[-1] not in (
-                        ALLOWED_TERMINATING_PUNCTUATIONS)
-                    no_word_is_present_in_excluded_phrases = not any(
-                        word in line for word in EXCLUDED_PHRASES)
-                    if last_char_is_invalid and (
-                            no_word_is_present_in_excluded_phrases):
-                        failed = True
-                        print '%s --> Line %s: %s' % (
-                            filename, line_num + 1, message)
-                        print ''
+        return summary_messages
 
-                # Check that comment starts with a space and is not a shebang
-                # expression at the start of a bash script which loses function
-                # when a space is added.
-                if space_regex.match(line) and not line.startswith('#!'):
-                    message = (
-                        'There should be a space at the beginning '
-                        'of the comment.')
-                    failed = True
-                    print '%s --> Line %s: %s' % (
-                        filename, line_num + 1, message)
-                    print ''
+    def _check_newline_character(self):
```

Comment:
Since #6172 went in, this function has become redundant. You can have a look at the referenced PR and take assurance for yourself. Also, it would help in your task of removing linter repetitions.

Reply:
Ok thanks for the info, removed it.